### ISSUE TITLE
including contributor e-mail addresses when their names aren't known

### ISSUE THREAD

**Issue by jay** [curl/curl#824]
Every once in a while I commit a patch from someone whose real name is unknown. Either I know their e-mail or their github name, which I turn into an e-mail the github recommended way `username@users.noreply.github.com`. For example recently I fixed a bug reported by github user jveazey and there is no real name associated with that profile. So in the commit message for 3caaeff I put `Reported-by: jveazey@users.noreply.github.com`. The contributor script has some post processing to "only count names with a space (ie more than one word)". Refer to https://github.com/curl/curl/blob/curl-7_49_0/scripts/contributors.sh#L61 .

So I would like to give these people proper credit. I think having them in the repo for the -by line is proper credit but then they aren't in the contributor list. Is that an oversight? I don't know how people with unknown real names would feel about having their e-mails in the contributor list (maybe easier for spam harvesting?) but in the case of github users we could put it in like `foo bar, jveazey (github), qux thud`. What do you think?

**Comment by bagder** [comment]
It is sort of made on purpose. If people don't provide a name, I see that as a sort of statement that they're not using a name and then they can't be referred to by name. People tend to be bit wary of having their email addresses showing up at places (for spam reasons) so simply inserting the full email might not be what these people would like either. Back in the day I used to email these people and ask them how they would like to be given credit for their help, but these days I often just let them not end up in the THANKS file. Right or wrong.

We similarly exclude aliases and nicknames. I've sort of gotten the idea that people use aliases and nicknames because they don't want to use their name so I've not added such names either. Also because aliases are even less unique than real names so the alias of today may not be the same person that used that alias a year ago...

For users at known sites like github, saying "jveazey at github" seems fine enough but I'm not sure how we'd deal with other domains...

    $ git log | grep by: | grep @ | cut -d: -f2   | grep -v '<'

This lists 12 users that are email only. 10 of them on github, so maybe we should just make the script handle github emails...

**Comment by jay** [comment]
so how about
~~~diff
diff --git a/scripts/contributors.sh b/scripts/contributors.sh
index 8838610..5ebac1c 100755
--- a/scripts/contributors.sh
+++ b/scripts/contributors.sh
@@ -50,7 +50,7 @@ cut -d: -f2- | \
 cut '-d<' -f1 | \
 tr , '\012' | \
 sed 's/ and /\n/' | \
-sed -e 's/^ //' -e 's/ $//g'
+sed -e 's/^ //' -e 's/ $//g' -e 's/@users.noreply.github.com$/ \(github\)/'

 grep "^  [^ \(]" RELEASE-NOTES| \
 sed 's/, */\n/g'| \
~~~
~~~
$ scripts/contributors.sh 46c4ad36edc
  Alexander Traud, Daniel Stenberg, Gisle Vanem, Jan Ehrhardt,
  jveazey (github), Marcel Raad, Michael Kaufmann, Paul Howarth, Ray Satiro,
  Steve Holme, Tomas Jakobsson,
  (11 contributors)
~~~

**Comment by bagder** [comment]
or possibly "jveazey on github" to avoid the parentheses.

**Comment by bagder** [comment]
~~~
$ ./scripts/contributors.sh curl-7_49_0
  Alexander Traud, Daniel Stenberg, Gisle Vanem, Jan Ehrhardt,
  jveazey on github, Marcel Raad, Michael Kaufmann, Paul Howarth, Ray Satiro,
  Steve Holme, Tomas Jakobsson,
  (11 contributors)

**Comment by jay** [comment]
Ok. Thanks, landed in 1577bfa.
